plasma_framework/test/src/transactions/outputs/PaymentOutputModel.test.js

```diff
@@ -0,0 +1,34 @@
+const rlp = require('rlp');
+const { constants, expectRevert } = require('openzeppelin-test-helpers');
+
+const PaymentOutputModelMock = artifacts.require("PaymentOutputModelMock");
+
+const PaymentTransactionOutput = require("../../../helpers/transaction.js").PaymentTransactionOutput;
+
+const OUTPUT_GUARD = "0x" + Array(64).fill(1).join("");
+
+contract("PaymentOutputModel", () => {
+
+    before(async () => {
+        this.test = await PaymentOutputModelMock.new();
+    });
+
+    it("should decode output", async () => {
+        const expected = new PaymentTransactionOutput(100, OUTPUT_GUARD, constants.ZERO_ADDRESS);
+        const encoded = web3.utils.bytesToHex(rlp.encode(expected.formatForRlpEncoding()));
+
+        const output = await this.test.decode(encoded);
+        const actual = decodeOutput(output);
+
+        expect(JSON.stringify(actual)).to.equal(JSON.stringify(expected));
+    });
+})
+
+function decodeOutput(output) {
```

Comment:
I have an idea on how to move this function under `PaymentTransactionOutput`. This would solve my comment below as well:

```js
class PaymentTransactionOutput {
    .....// existing codes

    static fromContractOutput(output) {
        // do things in your `decodeOutput(output)` function
        return new this(amount, outputGuard, token);
     }
}
```

and then we can use it directly:
```js
const actual = PaymentTransactionOutput.fromContractOutput(output);
```

Reply:
done